src/main/java/games/strategy/engine/pbem/TripleAForumPoster.java

```diff
@@ -77,32 +82,38 @@ private String uploadSaveGame(final CloseableHttpClient client, final String tok
         final String json = EntityUtils.toString(response.getEntity());
         return "\n[Savegame](" + new JSONArray(json).getJSONObject(0).getString("url") + ")";
       }
-      throw new Exception("Failed to upload savegame, server returned Error Code " + status);
+      throw new IllegalStateException("Failed to upload savegame, server returned Error Code " + status);
     }
   }
 
   private static void deleteToken(final CloseableHttpClient client, final int userId, final String token)
       throws IOException {
     final HttpDelete httpDelete = new HttpDelete(tripleAForumURL + "/api/v1/users/" + userId + "/tokens/" + token);
     addTokenHeader(httpDelete, token);
-    client.execute(httpDelete).close();
+    try (CloseableHttpResponse response = client.execute(httpDelete)) {
+      final int code = response.getStatusLine().getStatusCode();
+      if (code != HttpURLConnection.HTTP_OK) {
+        // Cleanup failed, this is not a severe issue
+        ClientLogger.logQuietly("Failed to delete API Token after posting, server responded with error code " + code);
+      }
+    }
   }
 
-  private int getUserId(final CloseableHttpClient client) throws Exception {
+  private int getUserId(final CloseableHttpClient client) throws IOException {
     final JSONObject jsonObject = login(client);
     checkUser(jsonObject);
     return jsonObject.getInt("uid");
   }
 
-  private static void checkUser(final JSONObject jsonObject) throws Exception {
+  private static void checkUser(final JSONObject jsonObject) {
     if (jsonObject.has("message")) {
-      throw new Exception(jsonObject.getString("message"));
+      throw new IllegalStateException(jsonObject.getString("message"));
     }
     if (jsonObject.getInt("banned") != 0) {
-      throw new Exception("Your account is banned from the forum");
+      throw new IllegalStateException("Your account is banned from the forum");
     }
     if (jsonObject.getInt("email:confirmed") != 1) {
-      throw new Exception("Your email isn't confirmed yet!");
+      throw new IllegalStateException("Your email isn't confirmed yet!");
```

Comment:
I'm not sure if IllegalStateException is the right exception type here.
I mean the exception type is not really important anyways, but this is a state that could occur if the user enters invalid credentials

Reply:
I think IllegealStateException is alright here.

Reply:
Beyond exception type, this could be a frustrating message for a user, particularly if they think their email is confirmed. What should someone do to confirm their email, how can they check? Those would be good details to let someone know rather than a more vague "your email is bad".

Lower value, better exception type for our logs, an "EmailNotConfirmed" exception would be appropriate. Such a custom exception could hide details of the error messaging.